### PR TITLE
Compile rust compiler from scratch

### PR DESCRIPTION
**Pull request by lrusak**
This changes the rust compiler to be built from scratch (after being bootstrapped by a pre-compiled compiler).

This allows us to tune the compiler a bit more and have more control over the entire build process.

Currently only librespot uses rust but there may be more packages in the future (https://github.com/systemd/zram-generator)

The problem here is that there is no opportunity to use ccache so it adds quite a bit of compile time. This also requires compiling llvm for the build machine as rust uses llvm. Rust isn't used in the default build though so it's not really a problem for now.

I've tested this for x86_64 and arm projects.

This uses slightly adjusted instructions from https://rustc-dev-guide.rust-lang.org/building/how-to-build-and-run.html
It is more similar to https://github.com/meta-rust/meta-rust

### CODE REVIEW

File: packages/lang/llvm/package.mk [hunk 1]
```diff
@@ -28,11 +27,10 @@ PKG_CMAKE_OPTS_COMMON="-DLLVM_INCLUDE_TOOLS=ON \
                        -DLLVM_ENABLE_SPHINX=OFF \
                        -DLLVM_ENABLE_OCAMLDOC=OFF \
                        -DLLVM_ENABLE_BINDINGS=OFF \
-                       -DLLVM_TARGETS_TO_BUILD=AMDGPU \
                        -DLLVM_ENABLE_TERMINFO=OFF \
                        -DLLVM_ENABLE_ASSERTIONS=OFF \
                        -DLLVM_ENABLE_WERROR=OFF \
-                       -DLLVM_ENABLE_ZLIB=ON \
+                       -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (CvH) [inline packages/lang/llvm/package.mk:33]: is this change needed to make it work or just some cleanup ?

Comment (lrusak) [reply packages/lang/llvm/package.mk:33]: It was required to get rust to link properly otherwise it would complain about missing zlib symbols. I didn't have time to look into it and figure zlib support probably wasn't needed anyways.

**Review by CvH — APPROVED** [review]
